### ISSUE TITLE
Typing issue with Annotated

### ISSUE THREAD

**Issue by AlexanderPodorov** [microsoft/pyright#7238]
**Describe the bug**
See example code snippet

```python
from dataclasses import dataclass
from typing import Annotated


@dataclass
class Test:
    pass


TestAnnonated = Annotated[Test, ...]


def test(a: type[Test]) -> None:
    return None


# Shows warning in Pylance v2024.2.1 (Pyright 1.1.348)
# No warning in Pylance v2023.12.1 (Pyright 1.1.336)
test(TestAnnonated)

# This works:
test(Test)
```

**VS Code extension or command-line**
Pylance v2024.2.1 (Pyright 1.1.348)

**Comment by erictraut** [comment]
Pyright's behavior is correct here. The special form `Annotated` is not compatible with `type` when used in a value expression. `Annotated` is intended to be used to be used in a type expression (an annotation), not in a value expression. Pyright previously didn't generate an error here as it should have. Mypy correct does. I've been working to make pyright [more conformant](https://htmlpreview.github.io/?https://github.com/python/typing/blob/main/conformance/results/results.html) with the official typing spec. 

You can also read more about this in [this typing forum thread](https://discuss.python.org/t/is-annotated-compatible-with-type-t/43898/41).
